src/qt/optionsdialog.cpp

```diff
@@ -293,12 +293,21 @@ void OptionsDialog::on_resetButton_clicked()
     }
 }
 
+
 void OptionsDialog::on_openBitcoinConfButton_clicked()
 {
     /* explain the purpose of the config file */
-    QMessageBox::information(this, tr("Configuration options"),
-        tr("The configuration file is used to specify advanced user options which override GUI settings. "
-           "Additionally, any command-line options will override this configuration file."));
+    int ret = QMessageBox::information(this, tr("Configuration options"),
+                                       tr("The configuration file is used to specify advanced user options which override GUI settings. "
+                                          "Additionally, any command-line options will override this configuration file."),
+                                       QMessageBox::Cancel | QMessageBox::Ok, QMessageBox::Ok);
+
+    switch (ret) {
+    case QMessageBox::Cancel:
+        return;
+    case QMessageBox::Ok:
+        break;
+    }
```

Comment:
Is it simpler
```suggestion
    if (ret == QMessageBox::Cancel) return;
```
?

Reply:
Or even better:
```
    if (ret != QMessageBox::Ok) return;
```

The latter variant fixes a bug which also is present in master: when the dialog is opened and the program is being shutdown (e.g., via the `Exit` action of the systray icon menu) a configuration file gets opened.

Reply:
> if (ret != QMessageBox::Ok) return;

That's a great suggestion. Updated `switch/case` to this format, in 25bee9251374fa78d19077f2fb5fa0bfc5d651a2